components/script_bindings/codegen/codegen.py

```diff
@@ -2672,27 +2697,28 @@ class CGList(CGThing):
     Generate code for a list of GCThings.  Just concatenates them together, with
     an optional joiner string.  "\n" is a common joiner.
     """
-    def __init__(self, children, joiner: str = "") -> None :
+    def __init__(self, children: list[CGThing], joiner: str = "") -> None:
```

Comment:
This should be an `Iterable[CGThings]` so that it can work with both lists and generators or other collections like the comment mentions that it can be generator as well (`Make a copy of the kids into a list, because if someone passes in a generator ...`).

This will also solve the problem with the type mismatch (i.e needing `: list[CGThing]` annotions in the callers that pass `list[CGGeneric]`), which only happens because list is immutable and therefore invariant with [respect to subtyping.](https://typing.python.org/en/latest/reference/generics.html#variance-of-generic-types)